lib/IDE/Refactoring.cpp

```diff
@@ -6640,17 +6767,80 @@ class AsyncConverter : private SourceEntityWalker {
       llvm_unreachable("Invalid top handler");
     }
 
+    // If ArgNames is not None, then the arguments to the completion handler
+    // were unwrapped using guard-let into variables with these names.
+    Optional<SmallVector<StringRef, 4>> ArgNames;
+
+    if (TopHandler.isAmbiguousCallToParamHandler(CE)) {
+      ArgNames.emplace();
+      // If the completion handler call was ambiguous, then all of its
+      // parameters were most likely of Optional type - otherwise, one of the
+      // parameters would always have to be `nil` or both an error and a success
+      // are passed back.
+      // Since the continuation doesn't take Optional values, we need to unwrap
+      // them.
+      unsigned I = 1;
+      for (auto Arg : Args) {
+        StringRef ArgName;
+        if (auto DRE = dyn_cast<DeclRefExpr>(Arg)) {
+          ArgName = DRE->getDecl()->getBaseIdentifier().str();
+        } else if (auto IIOE = dyn_cast<InjectIntoOptionalExpr>(Arg)) {
+          if (auto DRE = dyn_cast<DeclRefExpr>(IIOE->getSubExpr())) {
+            ArgName = DRE->getDecl()->getBaseIdentifier().str();
+          }
+        }
+        if (ArgName.empty()) {
+          if (Args.size() == 1) {
+            // We only have a single result. 'result' seems a resonable name.
+            ArgName = createUniqueName("result").str();
+          } else {
+            // We are returning a tuple. Name the result elements 'result' +
+            // index in tuple.
+            ArgName = createUniqueName("result" + std::to_string(I)).str();
+          }
+        }
+        ArgNames->push_back(ArgName);
+        OS << tok::kw_guard << ' ' << tok::kw_let << ' ' << ArgName << ' '
+           << tok::equal << ' ';
+        convertNode(Arg, /*StartOverride=*/CE->getArgumentLabelLoc(I),
```

Comment:
The `I` used by the loop starts on 1, but this usage expects it to start on 0. I don't *think* it currently causes issues here as the handler being called should be a function value that cannot have argument labels (and the presence of the error param means we won't run off the end), but could this still be corrected in case argument labels are supported in the future?

Reply:
Changed `I` to start at 0, which matches better with tuple indexing anyway.